content/operate/kubernetes/security/vault.md

```diff
@@ -0,0 +1,530 @@
+
+   Configure a Vault role that binds the operator's service account to the policy:
+
+   ```bash
+   vault write -namespace=<VAULT_NAMESPACE> auth/<AUTH_PATH>/role/redis-enterprise-operator-<K8S_NAMESPACE> \
+           bound_service_account_names="redis-enterprise-operator" \
+           bound_service_account_namespaces=<K8S_NAMESPACE> \
+           policies=redisenterprise-<K8S_NAMESPACE>
+   ```
+
+   Parameter explanation:
+   - `<AUTH_PATH>`: Kubernetes auth method path in Vault (default: `kubernetes`)
+   - Role name includes namespace for multi-tenant isolation
+
+2. Configure operator environment
+
+   Create a ConfigMap with Vault configuration for the Redis Enterprise operator:
+
+   ```yaml
+   # operator-environment-config.yaml
+   apiVersion: v1
+   kind: ConfigMap
+   metadata:
+     name: operator-environment-config
+     namespace: <K8S_NAMESPACE>
+   data:
+     CREDENTIAL_TYPE: "vault"
+     VAULT_SERVER_FQDN: "<VAULT_FQDN>"
+     VAULT_SERVICE_PORT_HTTPS: "8200"
+     VAULT_SECRET_ROOT: "secret"
+     VAULT_SECRET_PREFIX: "redisenterprise-<K8S_NAMESPACE>"
+     VAULT_ROLE: "redis-enterprise-operator-<K8S_NAMESPACE>"
+     VAULT_AUTH_PATH: "<AUTH_PATH>"
+     VAULT_NAMESPACE: "<VAULT_NAMESPACE>"
+     VAULT_CACHE_SECRET_EXPIRATION_SECONDS: "120"
+   ```
+
+   Apply the configuration:
+
+   ```bash
+   kubectl apply -f operator-environment-config.yaml
+   ```
+
+   Configuration parameters:
+
+   | Parameter | Description | Default | Required |
+   |-----------|-------------|---------|----------|
+   | `CREDENTIAL_TYPE` | Must be set to `"vault"` to enable Vault integration | - | Yes |
+   | `VAULT_SERVER_FQDN` | Vault server hostname (e.g., `vault.vault-ns.svc.cluster.local`) | - | Yes |
+   | `VAULT_SERVICE_PORT_HTTPS` | Vault HTTPS port | `8200` | Yes |
+   | `VAULT_SECRET_ROOT` | KV-v2 secret engine mount path | `secret` | Yes |
+   | `VAULT_SECRET_PREFIX` | Prefix for all Redis Enterprise secrets | `redisenterprise` | Yes |
+   | `VAULT_ROLE` | Vault role for operator authentication | `redis-enterprise-operator` | Yes |
+   | `VAULT_AUTH_PATH` | Kubernetes auth method path | `kubernetes` | Yes |
+   | `VAULT_NAMESPACE` | Vault Enterprise namespace | - | Enterprise only |
+   | `VAULT_CACHE_SECRET_EXPIRATION_SECONDS` | Secret cache duration | `120` | No |
+
+   Secret path construction: Secrets are stored at `<VAULT_SECRET_ROOT>/data/<VAULT_SECRET_PREFIX>/<secret-name>`
+
+   <br>
+
+3. Deploy the operator
+
+   Deploy the Redis Enterprise operator following the [standard installation guide]({{< relref "/operate/kubernetes/deployment" >}}).
+
+   {{<warning>}}
+   The operator pod will not be ready until the admission controller secret is stored in Vault (covered in the next step).
+   {{</warning>}}
+
+   <br>
+
+4. Configure admission controller secret
+
+   Generate and store the admission controller TLS certificate in Vault:
+
+   ```bash
```

Comment:
Same thing with this code block and the one after the one below it.
<img width="670" height="696" alt="Screenshot 2025-11-14 at 14 12 29" src="https://github.com/user-attachments/assets/f8b1d853-54d4-424b-929b-479ce295c754" />